doc/architectural/CBMC-developer-guide.md

```diff
@@ -1,11 +1,196 @@
 \ingroup module_hidden 
-\page cprover-architecture-overview CProver Architecture Overview
+\ingroup module_hidden 
```

Comment:
I'd split this into separate .md files. If you give them appropriate \page directives they will appear as pages within the `CProver Developer Documentation` section.

Reply:
Perhaps when this file starts getting too big I will split of large parts into their own pages and just link to them from this page. I like having the headings in this order, as I think this is roughly the order that a new developer would want to encounter the documentation.